code/game/objects/items/devices/multitool.dm

```diff
@@ -49,10 +55,26 @@
 	src.buffer = buffer
 	RegisterSignal(buffer, COMSIG_QDELETING, PROC_REF(on_buffer_del))
 
+/**
+ * Called when the buffer's stored object is deleted
+ *
+ * This proc does not clear the buffer of the multitool, it is here to
+ * handle the deletion of the object the buffer references
+ */
 /obj/item/multitool/proc/on_buffer_del(datum/source)
 	SIGNAL_HANDLER
 	buffer = null
 
+/**
+ * Clears the multitool buffer
+ *
+ * Does not delete the object stored in the buffer,
+ * clears the buffer so a new object can be referenced
+ */
+/obj/item/multitool/proc/clear_buffer()
```

Comment:
This proc is redundant. set_buffer() already unregisters the signal and can also clear the buffer if you pass null to it. Just add a null check there to ensure we don't register a signal on a deleted object. Something like

 ```
/obj/item/multitool/proc/set_buffer(datum/buffer)
	if(src.buffer)
		UnregisterSignal(src.buffer, COMSIG_QDELETING)
	src.buffer = buffer
    if(!QDELETED(buffer))
	     RegisterSignal(buffer, COMSIG_QDELETING, PROC_REF(on_buffer_del))
```

Also I remember a lot of things passing null to set_buffer() anyway so this will make things a lot easier.

Reply:
Awesome, I didn't quite understand how the signals all worked together in this proc, but after looking at the better solution to this issue I get what's going on. Should be cleaner now.